addons/account/views/account_journal_dashboard_view.xml

```diff
@@ -298,9 +298,16 @@
                                 </button>
                             </t>
                             <t t-if="journal_type == 'purchase'">
-                                <button class="btn btn-primary o_button_upload_bill oe_kanban_action_button" journal_type="purchase" groups="account.group_account_invoice">
-                                    <span>Upload</span>
-                                </button>
+                                <t t-if="dashboard.number_draft > 0 || dashboard.number_waiting > 0 || dashboard.number_late > 0">
```

Comment:
If you have posted and paid all the invoices, it doesn't mean you want to have the onboarding step again.
Add a new computed field on `account_journal` using a `read_group` to count entries in each journal.

Reply:
After reading the documentation on the ORM API Search/Read methods and I found the `Model.search_count()` method which seemed to be appropriate in this case.

I did some testing and the results look OK to me. I hope it's alright :wink: